src/dynsight/_internal/utilities/utilities.py

```diff
@@ -16,3 +19,42 @@ def normalize_array(x: np.ndarray) -> np.ndarray:  # type: ignore[type-arg]
     norm = np.linalg.norm(x, axis=-1, keepdims=True)
     norm[norm == 0] = 1
     return x / norm
+
+
+def find_extrema_points(
+    x_axis: np.ndarray[float, Any],
+    y_axis: np.ndarray[float, Any],
+    extrema_type: str,
+    prominence: float,
+) -> np.ndarray[float, Any]:
+    """Find the extrema points of a mathematical function (minima or maxima).
+
+    Parameters:
+        x_axis:
+            x values of the function.
+        y_axis:
+            y values of the function.
+        extrema_type:
+            It can be "min" or "max" depending on what the user's choice.
+        prominence:
+            Required prominence of peaks. Higher values will provides only
+            well defined and sharp peaks excluding the softer ones.
+
+    Returns:
+        A NumPy array with dimensions (n_peaks, 2), containing
+        the x and y coordinates for each peak.
+    """
+    if extrema_type not in {"min", "max"}:
+        type_msg = "extrema_type must be 'min' or 'max'"
+        raise ValueError(type_msg)
+    if extrema_type == "min":
```

Comment:
For simplicity, I would write it like this:
```python
if extrema_type == 'min':
    function = -y_axis
else:
    function = y_axis

peaks, properties = find_peaks(x=function, prominence=prominence)
```

Reply:
Although be careful to make sure this is not creating a point -- if `y_axis` is a list, you could be messing up the variable in this way

Reply:
added a test for the critical points research